File: Algorithms/2Fast2Furious.cpp
//https://www.hackerearth.com/practice/basic-programming/implementation/basics-of-implementation/practice-problems/algorithm/2-fast-2-furious/?utm_campaign=weekly_hackathon_reco&utm_medium=email&utm_source=recommendation&utm_content=hackathon
#include<iostream>
#include<stdlib.h>
using namespace std;


int main()
    {
        long int n,i,a,temp,dom,brian;
        a=temp=dom=brian=0;
        cin>>n;
        for(i=0;i<n;i++)
            {
                if(abs(temp-a)>dom)
                    dom=abs(temp-a);
                temp=a;
                cin>>a;
                
            }
        if(abs(temp-a)>dom)
            dom=abs(temp-a);
            
        temp=a=0;
        for(i=0;i<n;i++)
            {
                if(abs(temp-a)>brian)
                    brian=abs(temp-a);
                temp=a;
                cin>>a;
                
            }
        if(abs(temp-a)>brian)
            brian=abs(temp-a);
            
        if(brian>dom)
            {
            cout<<"Brian"<<endl;
            cout<<brian;
            }
        else if(brian<dom)
            {
            cout<<"Dom\n";
            cout<<dom;
            }
        else
            {
            cout<<"Tie\n";
            cout<<dom;
            }
            
        
        return 0;
    }

File: Algorithms/2Fast2Furious.py
'''
# Sample code to perform I/O:

name = input()                  # Reading input from STDIN
print('Hi, %s.' % name)         # Writing output to STDOUT

# Warning: Printing unwanted or ill-formatted data to output will cause the test cases to fail
'''
#https://www.hackerearth.com/practice/basic-programming/implementation/basics-of-implementation/practice-problems/algorithm/2-fast-2-furious/?utm_campaign=weekly_hackathon_reco&utm_medium=email&utm_source=recommendation&utm_content=hackathon
# Write your code here
n = int(input())
temp,a,brian,dom=0,0,0,0

x = input().split(" ")
y = input().split(" ")

for i in range(n):
    if abs(temp-int(x[i])) > dom :
        dom = abs(temp-int(x[i]))
    temp = int(x[i])

if abs(temp-int(x[i])) > dom :
    dom = abs(temp-int(x[i]))
    
temp,a = 0,0
for i in range(n):
    if abs(temp-int(y[i])) > brian :
        brian = abs(temp-int(y[i]))
    temp = int(y[i])
    
if abs(temp-int(y[i])) > brian :
    brian = abs(temp-int(y[i]))
    
if brian > dom:
    print("Brian\n")
    print(brian)
elif brian < dom :
    print("Dom\n")
    print(dom)
else :
    print("Tie\n")
    print(dom)
    
